crates/valence/src/inventory/validate.rs

```diff
@@ -0,0 +1,643 @@
+use valence_protocol::packet::c2s::play::click_slot::ClickMode;
+use valence_protocol::packet::c2s::play::ClickSlotC2s;
+
+use super::{Inventory, InventoryWindow};
+use crate::prelude::CursorItem;
+
+/// Validates a click slot packet enforcing that all fields are valid.
+pub(crate) fn validate_click_slot_impossible(
```

Comment:
Can this return an `anyhow::Result<()>` instead of `bool`? Would help with debugging. Same goes for `validate_click_slot_item_duplication`.